diff -urN ExtUtils-F77-1.16.orig/F77.pm ExtUtils-F77-1.16/F77.pm
--- ExtUtils-F77-1.16.orig/F77.pm	2008-05-11 13:53:09.000000000 +0200
+++ ExtUtils-F77-1.16/F77.pm	2008-05-11 13:57:10.000000000 +0200
@@ -194,7 +194,7 @@
         $dir = "/usr/local/lib";
         $lib = "f2c";
     }    
-    return( "-L$dir -L/usr/lib -l$lib -lm" );
+    return( "-L$dir -L/usr/lib -L/usr/lib64 -l$lib -lm" );
 };
 $F77config{Generic}{G77}{Trail_} = 1;
 $F77config{Generic}{G77}{Compiler} = find_in_path('g77','f77','fort77');
@@ -237,7 +237,7 @@
     } else {
         $dir = "/usr/local/lib";
     }    
-    return( "-L$dir -L/usr/lib -lgfortran -lm" );
+    return( "-L$dir -L/usr/lib -L/usr/lib64 -lgfortran -lm" );
 };
 $F77config{Linux}{GFortran}{Trail_}   = 1;
 $F77config{Linux}{GFortran}{Compiler} = 'gfortran';
@@ -565,7 +565,7 @@
    # Create list of directories to search (with common defaults)
 
    my @path = ();     
-   for (@args, "/usr/lib", "/lib") { 
+   for (@args, "/usr/lib", "/lib", "/usr/lib64", "/lib64") { 
       push @path, $1 if /^-L(.+)$/ && -d $1;
    }
 
